packages/plugin-secure-proxy/src/index.ts

```diff
@@ -32,6 +53,27 @@ export function pluginSecureProxy(pluginParams: SecureProxyPluginOptions): Orama
       proxy,
       pluginParams
     },
+
+    async beforeInsert<T extends TypedDocument<any>>(_db: AnyOrama, _id: string, params: PartialSchemaDeep<T>) {
+      if (!pluginParams.embeddings?.onInsert?.generate) {
```

Comment:
embedding is not optional, right? so we can remove the `?` after `embeddings` (here and below)

Reply:
That's right, `embeddings` is not optional. Thanks!